### DISCUSSION TITLE
Vercel Server environment variables being populated

### DISCUSSION THREAD

**Discussion by jacobsamo** [vercel/community#7311] [Help]
### Pages to Investigate

https://github.com/t3-oss/t3-env
https://github.com/jacobsamo/Mixie/tree/dev

### Steps to Reproduce
I am currently using using [T3 Env](https://env.t3.gg/) for type safe environment variables however on vercel it seems that the server environment variables aren't pulling in for some reason. it builds fine in locally and builds fine using the Vercel CLI using `vercel pull` and `vercel build`. 

I have checked that my environment variables are correct in my settings and they are and are set for the right environments. 
This is the error I see:
```bash
mixie/website:build: $ next build
@mixie/website:build: ❌ Invalid environment variables: {
@mixie/website:build:   SUPABASE_URI: [ 'Required' ],
@mixie/website:build:   UNSPLASH_SECRET: [ 'Required' ],
@mixie/website:build:   UPLOADTHING_SECRET: [ 'Required' ],
@mixie/website:build:   UPLOADTHING_APP_ID: [ 'Required' ],
@mixie/website:build:   SUPABASE_SERVICE_ROLE: [ 'Required' ],
@mixie/website:build:   GOOGLE_AI_API_KEY: [ 'Required' ],
@mixie/website:build:   UPSTASH_REDIS_REST_URL: [ 'Required' ],
@mixie/website:build:   UPSTASH_REDIS_REST_TOKEN: [ 'Required' ]
@mixie/website:build: }
@mixie/website:build:  ⨯ Failed to load next.config.mjs, see more info here https://nextjs.org/docs/messages/next-config-error
@mixie/website:build: 
@mixie/website:build: > Build error occurred
@mixie/website:build: Error: Invalid environment variables
@mixie/website:build:     at onValidationError (file:///vercel/path0/node_modules/@t3-oss/env-core/dist/index.js:29:15)
@mixie/website:build:     at createEnv (file:///vercel/path0/node_modules/@t3-oss/env-core/dist/index.js:35:16)
@mixie/website:build:     at createEnv (file:///vercel/path0/node_modules/@t3-oss/env-nextjs/dist/index.js:12:12)
@mixie/website:build:     at file:///vercel/path0/apps/website/env.mjs:4:20
@mixie/website:build:     at ModuleJob.run (node:internal/modules/esm/module_job:222:25)
@mixie/website:build:     at async ModuleLoader.import (node:internal/modules/esm/loader:323:24)
@mixie/website:build:     at async loadConfig (/vercel/path0/node_modules/next/dist/server/config.js:698:36)
```

### What else have I tried
I tired removing the package and running with normal `process.env` however still no luck. environment variables are still not being populated

the repo can be found here: https://github.com/jacobsamo/Mixie/tree/dev

**Comment by TobeBetterV** [comment]
Are you still troubled by this issue? The last link of your problem details is private.

**Reply by jacobsamo** [reply]
I am still experiencing the issue, Just changed the links but here is the link to `dev` as it seems to be happening on all environments https://github.com/jacobsamo/Mixie/tree/dev

**Comment by jacobsamo** [comment]
Moved to discussion: https://github.com/orgs/vercel/discussions/7460 which provides more detail and description

[a comment by an automated account is not shown]
